### ISSUE TITLE
Unintentional Download of Coco Dataset: Training and Validation Process

### ISSUE THREAD

**Issue by HosseinShahverdi** [ultralytics/ultralytics#14979]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

![code_1](https://github.com/user-attachments/assets/7894c06a-1edb-489a-a84e-89b387b8ef8d)
![code_2](https://github.com/user-attachments/assets/52f5fd4c-1c32-4cfc-8965-9efe5853366d)

I'm trying to train YOLOv8 on my own dataset. However, when I run the code (attached snapshot), the model automatically attempts to download coco.yaml and starts training on the COCO dataset for both detection and segmentation tasks.

How can I prevent YOLO from using the COCO dataset and ensure it reads the dataset path specified in my own YAML file?

Thanks for your help!

**Comment by Y-T-G** [comment]
That looks like GPT code. That's not how you train a YOLOv8 model. 

https://docs.ultralytics.com/modes/train/

**Comment by HosseinShahverdi** [comment]
> That looks like GPT code. That's not how you train a YOLOv8 model.
> 
> https://docs.ultralytics.com/modes/train/

Hello, thank you for your response. In fact, I did not use AI to generate it; I wrote it myself. could you please explain to me in which part I make mistake that yolo tries to train coco

**Comment by Y-T-G** [comment]
It wont work. There's no loss function. And the YOLO `model` is not a typical PyTorch model where `model.train()` or `model.val()` works.
